### PR TITLE
Midi reconstruction

### PR DESCRIPTION
**Pull request by jrmsgr**
Added functions in the utils folder to convert a Staff object into a MIDI file.
For a detailed example, check examples/midi_encoding. Don't hesitate to play the resulting file, result.mdi !

### CODE REVIEW

File: music_detection/note.py [hunk 1]
```diff
@@ -2,8 +2,13 @@
...
 class Note:
-    def __init__(self):
-        self.name = NoteEnum.UNDEFINED
-        self.octave = 0
-        self.duration = 0
+    def __init__(self, *args):
+        if (len(args)> 0) :
+            self.name = args[0]
+            self.octave= args[1]
+            self.duration = args[2]
```
Comment (Ratoone) [inline music_detection/note.py]: Interesting choice of constructor, but what about a constructor with default values?

File: music_detection/utils/midi_functions.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+from music_detection.staff import Staff
+from music_detection.note import Note
+from music_detection.time_enum import TimeSignatureEnum
+from music_detection.note_enum import NoteEnum
+
+import midiutil as mu
+
+def get_time_signature_values(staff : Staff) -> int :
```
Comment (Ratoone) [inline music_detection/utils/midi_functions.py]: I would suggest passing only the time signature, as you don't need the whole staff object for this

File: music_detection/utils/midi_functions.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    elif name==NoteEnum.LA :
+        note_offset = 9
+    elif name==NoteEnum.SI :
+        note_offset = 11
+    
+    MIDIValue = (octave+1)*12+note_offset
+    return MIDIValue
+
+#TODO : MAYBE change the following function so that instead of a unique Staff object it takes
+#        a list of them as input
```
Comment (Ratoone) [inline music_detection/utils/midi_functions.py]: Either that, or have a class MidiWriter, and after initialization you append staff by staff, exporting the file only after the last one

**Review by Ratoone — CHANGES_REQUESTED** [review]

File: music_detection/utils/midi_functions.py (hunk 3 above)
Comment (jrmsgr) [reply music_detection/utils/midi_functions.py]: I did as you told and created a MIDIWriter class that takes in charge all the process

File: music_detection/utils/midi_functions.py (hunk 2 above)
Comment (jrmsgr) [reply music_detection/utils/midi_functions.py]: I did the modification, check it out

File: music_detection/note.py (hunk 1 above)
Comment (jrmsgr) [reply music_detection/note.py]: Modified the constructor so that it has default values

**Review by Ratoone — APPROVED** [review]

**Pull request merged**